### ISSUE TITLE
'terraform import' is stripping last character of resource name

### ISSUE THREAD

**Issue by mig5** [hashicorp/terraform#16273]
### Terraform Version

v0.10.7

### Terraform Configuration Files

Config defining the module:

```
module "database-encrypted" {
  source               = "./modules/database-encrypted"
  engine               = "mysql"
  engine_version       = "5.6.35"
  mysql_user           = "produser"
  mysql_pass           = "XXXXXXX"
  mysql_size           = "db.t2.large"
  allocated_storage    = "40"
  backup_retention     = "14"
  multi_az             = "true"
  storage_type         = "gp2"
  subnet_id_a          = "${module.vpc.prod_vpc_subnet_a}"
  subnet_id_b          = "${module.vpc.prod_vpc_subnet_b}"
  subnet_id_c          = "${module.vpc.prod_vpc_subnet_c}"
  vpc_id               = "${module.vpc.prod_vpc_id}"
  cidr_block           = "${module.vpc.prod_vpc_cidr_block}"
  parameter_group_name = "default.mysql5.6"
  mysql_access         = ["${module.vpc.prod_vpc_cidr_block}"]
}
```

Main bit of modules/database-encrypted/ code:

```
resource "aws_db_instance" "prod-db-encrypted" {
  identifier              = "prod-db-encrypted"
  engine                  = "${var.engine}"
  engine_version          = "${var.engine_version}"
  instance_class          = "${var.mysql_size}"
  username                = "${var.mysql_user}"
  password                = "${var.mysql_pass}"
  db_subnet_group_name    = "${aws_db_subnet_group.prod_encrypted.name}"
  parameter_group_name    = "${var.parameter_group_name}"
  vpc_security_group_ids  = ["${aws_security_group.prod_db_encrypted.id}"]
  multi_az                = "${var.multi_az}"
  apply_immediately       = "true"
  backup_retention_period = "${var.backup_retention}"
  allocated_storage       = "${var.allocated_storage}"
  storage_type            = "${var.storage_type}"
  skip_final_snapshot     = "false"

  tags {
    Name = "Prod RDS"
  }
}
```

Tried to run:

```
terraform import module.database-encrypted.prod-db-encrypted db-XXXXXXXXXXXXXXXXX
```

### Expected Behavior

I expected to be able to import a running RDS instance into this state

### Actual Behavior

```
Error: module.database-encrypte does not exist in the configuration.

Please add the configuration for the module before importing resources into it.
```

Note the last 'd' in 'database-encrypted' has been stripped.

It doesn't matter how long the string name is - I tried renaming both the module and the resource to 'database-enc' and it still stripped one character off. 

The issue also occurred in the previous release v0.10.6. The last time I recall trying it and it working was v0.9.11.

**Comment by apparentlymart** [comment]
Hi @mig5! Sorry for this confusing outcome.

I think this is actually just a misleading error message; the address specified isn't valid, but rather than detecting and reporting that it's instead misinterpreting it.

The problem is that the module address must include the resource type as well as its name, like this:

```
terraform import module.database-encrypted.aws_db_instance.prod-db-encrypted db-XXXXXXXXXXXXXXXXX
```

**Comment by mig5** [comment]
@apparentlymart thanks for clarification, I confirm that worked!

**Comment by apparentlymart** [comment]
Great, @mig5! I'll leave this open to remember that there's a bug in the resource address parsing here, where it's not correctly catching this error. We will probably rewrite the resource address parser soon, as part of a sequence of work to revamp the configuration language... that will probably take care of this issue as a side-effect, but we can leave this here to make sure.

**Comment by apparentlymart** [comment]
Hi again @mig5!

I just tried your original command line again in Terraform v0.12.0-alpha2 to confirm, and it is indeed fixed by the new resource address parser, which can produce better errors:

![Error: Invalid address; Resource specification must include a resource type and name.](https://user-images.githubusercontent.com/20180/48746000-74f42280-ec22-11e8-83ef-3ff6741cc630.png)

The formatting of the message is a little strange because the error is coming from a non-file source, but it does its best to render the relevant part of the input and describe what it was expecting.

Since this is already in the master branch and ready to be included in the forthcoming v0.12.0 release, I', going to close this out now. Thanks for reporting this, and sorry for the delay in getting it fixed!
